app/src/main/java/com/android/unio/ui/event/EventDetails.kt

```diff
@@ -608,40 +610,34 @@ fun EventSaveButton(event: Event, eventViewModel: EventViewModel, userViewModel:
           }
         }
       }
-
+  var enableButton by remember { mutableStateOf(true) }
+  val isConnected = NetworkUtils.checkInternetConnection(context)
   val onClickSaveButton = {
-    if (isSaved) {
-      val newEvent = event.copy(numberOfSaved = event.numberOfSaved - 1)
-      eventViewModel.updateEventWithoutImage(
-          newEvent,
-          onSuccess = {},
-          onFailure = { e -> Log.e("EventCard", "Failed to update event: $e") })
-      userViewModel.unsaveEvent(event) {
-        if (notificationPermissionsEnabled) {
-          NotificationWorker.unschedule(context, event.uid.hashCode())
-        }
-      }
-      Firebase.messaging.unsubscribeFromTopic(event.uid)
-    } else {
-      val newEvent = event.copy(numberOfSaved = event.numberOfSaved + 1)
-      eventViewModel.updateEventWithoutImage(
-          newEvent,
-          onSuccess = {},
-          onFailure = { e -> Log.e("EventCard", "Failed to update event: $e") })
-      userViewModel.saveEvent(event) {
-        if (!notificationPermissionsEnabled) {
-          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
-            // this permission requires api 33
-            // We should check how to make notifications work with lower api versions
-            permissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS)
+    enableButton = false
+    if (isConnected) {
+      eventViewModel.updateSave(events.first { it.uid == event.uid }, user!!, isSaved) {
+        userViewModel.refreshUser()
+        if (isSaved) {
+          if (notificationPermissionsEnabled) {
+            NotificationWorker.unschedule(context, event.uid.hashCode())
           }
         } else {
-          scheduleReminderNotification()
+          if (!notificationPermissionsEnabled) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+              // this permission requires api 33
+              // We should check how to make notifications work with lower api versions
+              permissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS)
+            }
+          } else {
+            scheduleReminderNotification()
+          }
```

Comment:
How did you catch that ?? Wonderful

Reply:
Not me sadly, it was already in the previous code :cry: